docs/pages/onBeforeRender.page.server.mdx

```diff
@@ -55,6 +55,36 @@ This is where you usually pass information about the authenticated user
 > ```
 
 
+## TypeScript
+
+```ts
+export { onBeforeRender }
+
+import type { OnBeforeRenderAsync } from 'vike/types'
+
+const onBeforeRender: OnBeforeRenderAsync = async (
+  pageContext
+): ReturnType<OnBeforeRenderAsync> => {
+  const response = await fetch("https://api.imdb.com/api/movies/")
+  const { movies } = await response.json()
+  return {
+    pageContext: {
+      movies
+    }
+  }
+}
+```
+
+> Do not omit the `ReturnType<OnBeforeRenderAsync>` annotation (i.e. do not write
+> `const onBeforeRender: OnBeforeRenderAsync = async (pageContext) => {`), otherwise TypeScript won't strictly check
+> the return type for unknown extra properties. See also:
+> * https://stackoverflow.com/questions/63006497/typescript-why-not-strict-return-type
+> * https://stackoverflow.com/questions/57214325/typescript-excess-properties-are-not-checked-in-lambdas-without-explicit-return
+
+> See <Link href="/pageContext#typescript" /> for more information on how to extend `pageContext` with your own extra
+> properties.
```

Comment:
@brillout what do you think of this?

Of course I'll move this text to a separate component so I don't copy-paste it across hook docs.

Reply:
👍